projects/motor_controller_interface/src/motor_controller_fsm.c

```diff
@@ -0,0 +1,321 @@
+    // rather than backwards
+    desired_direction = MOTOR_CONTROLLER_FORWARD_VELOCITY_MPS;
+  }
+
+  // To run the motor in torque control mode, set the velocity to an
+  // unobtainable value such as 100 m/s. Set the current to a value that is
+  // proportional to your accelerator pedal position.
+  MotorControllerDriveCmd cmd = {
+    .current_percentage = desired_torque,
+    .velocity = desired_direction,
+  };
+
+  // Update Left Motor Controller message
+  GenericCanMsg left_msg = {
+    .id = MOTOR_CONTROLLER_DRIVE_COMMAND_ID(MOTOR_CONTROLLER_INTERFACE_LEFT_ADDR),
+    .data = 0,
+    .dlc = sizeof(cmd),
+    .extended = false,
+  };
+  memcpy(&left_msg.data, &cmd, sizeof(cmd));
+  s_left_interval->msg = left_msg;
+
+  // Update Right Motor Controller message
+  GenericCanMsg right_msg = {
+    .id = MOTOR_CONTROLLER_DRIVE_COMMAND_ID(MOTOR_CONTROLLER_INTERFACE_RIGHT_ADDR),
+    .data = 0,
+    .dlc = sizeof(cmd),
+    .extended = false,
+  };
+  memcpy(&right_msg.data, &cmd, sizeof(cmd));
+  s_right_interval->msg = right_msg;
+
+  // Trigger a transmission
+  can_interval_send_now(s_left_interval);
+  can_interval_send_now(s_right_interval);
+}
+
+static void prv_cruise_control_mode(FSM *fsm, const Event *e, void *context) {
+  DriverControlsData data = { 0 };
+  fifo_pop(s_fifo, &data);
+  float desired_speed = MOTOR_CONTOLLER_VELOCITY_CMPS_TO_MPS(data.cruise_control);
+
+  // We perform cruise control by setting velocity mode on one motor, then
+  // copying the current setpoint to the other motor
+
+  // To run the motor in velocity (cruise) control mode, set the current to your
+  // maximum desired acceleration force (usually 100%), and set the velocity to
+  // the desired speed.
+  MotorControllerDriveCmd cmd_left = {
+    .current_percentage = 1.0f,
+    .velocity = desired_speed,
+  };
+
+  // Update Left Motor Controller message
+  GenericCanMsg left_msg = {
+    .id = MOTOR_CONTROLLER_DRIVE_COMMAND_ID(MOTOR_CONTROLLER_INTERFACE_LEFT_ADDR),
+    .data = 0,
+    .dlc = sizeof(cmd_left),
+    .extended = false,
+  };
+  memcpy(&left_msg.data, &cmd_left, sizeof(cmd_left));
+  s_left_interval->msg = left_msg;
+
+  // Read the current setpoint from the left Motor Controller and send it to
+  // the Right Motor Controller
+  MotorControllerPowerCmd cmd_right = {
+    .current_percentage = s_mc_measurements->bus_meas_left.current / MOTOR_CONTROLLER_CURRENT_LIMIT,
+  };
+  GenericCanMsg right_msg = {
+    .id = MOTOR_CONTROLLER_POWER_COMMAND_ID(MOTOR_CONTROLLER_INTERFACE_RIGHT_ADDR),
+    .data = 0,
+    .dlc = sizeof(cmd_right),
+    .extended = false,
+  };
+  memcpy(&right_msg.data, &cmd_right, sizeof(cmd_right));
+  s_right_interval->msg = right_msg;
+
+  // Trigger a transmission
+  can_interval_send_now(s_left_interval);
+  can_interval_send_now(s_right_interval);
+}
+
+StatusCode motor_controller_fsm_init(const MotorControllerFsmStorage *storage) {
+  status_ok_or_return(
```

Comment:
Couldn't you just use a loop for this?

Reply:
Yeah, I guess so.

I actually had a `TODO` here that I removed because I didn't have an associated JIRA, but it was something along the lines of

```c
// Either clean this up or do some clever masking to reduce 
// the number of RX handlers needed
```

I'm pretty sure you can reduce the amount of handlers needed if we make use of the masks, and then disable the other telemetry messages we don't care about via the WaveSculptor config tool.

But yeah, I guess I can also just allocate an array of all the message IDs that I care about, and then iterate through those.

Reply:
Yeah, I think masking would be a cleaner solution (especially if we return early from our software filters)